lib/propagation.js

```diff
@@ -1,25 +1,33 @@
+// @flow
+import {ITERATION_WIDTH} from './constants'
+import {getNonTransitTime} from './utils'
+
+import type {Origin, Query, StopTreeCache} from './types'
+
 /**
- * Perform propagation from an origin to destination cells, calling a callback to summarize the data.
- *
- * This seems like it's overly complicated, but I don't think there's an easier way to have this code
- * used both to create surfaces/isochrones and spectrogram data, which are different summaries of the data.
+ * Perform propagation from an origin to destination cells, calling a callback
+ * to summarize the data. This seems like it's overly complicated, but I don't
+ * think there's an easier way to have this code used both to create surfaces,
+ * isochrones and spectrogram data, which are different summaries of the data.
  *
- * The callback will be called with the following, once for each destination pixel
+ * The callback will be called once for each destination pixel with:
  *   travelTimesForDest: Travel times at each iteration
  *   walkTimesForDest: Walk times at each iteration
  *   inVehicleTravelTimesForDest: In vehicle travel times at each iteration
  *   waitTimesForDest: Wait times at each iteration
  *   x: X coordinate of grid for this destination
  *   y: Y coordinate of grid for this destination
  */
-
-import {getNonTransitTime, ITERATION_WIDTH} from './origin'
-
 export default function propagate ({
   next,
   origin,
   query,
   stopTreeCache
+}: {
+  next: Function,
```

Comment:
[The Function type is unsafe and should be avoided](https://flow.org/en/docs/types/functions/#toc-function-type).